metadata-ingestion/src/datahub/ingestion/source/gc/datahub_gc.py

```diff
@@ -140,36 +161,61 @@ def create(cls, config_dict, ctx):
     def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         return [partial(auto_workunit_reporter, self.get_report())]
 
+    @contextmanager
```

Comment:
You van use IngestionReporter for this -> https://github.com/datahub-project/datahub/blob/d5e05131d5dbcb37b62e4218dac83999d7b45bc1/metadata-ingestion/src/datahub/ingestion/source_report/ingestion_stage.py#L22

Here is an example usage -> 
Import to report
https://github.com/datahub-project/datahub/blob/d5e05131d5dbcb37b62e4218dac83999d7b45bc1/metadata-ingestion/src/datahub/ingestion/source/unity/report.py#L22

Here is the usage:
https://github.com/datahub-project/datahub/blob/d5e05131d5dbcb37b62e4218dac83999d7b45bc1/metadata-ingestion/src/datahub/ingestion/source/unity/source.py#L266

Reply:
I will look into this. Thanks for letting me know